pkg/metrics/metrics.go

```diff
@@ -0,0 +1,232 @@
+package metrics
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/kanisterio/kanister/pkg/log"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"gonum.org/v1/gonum/stat/combin"
+)
+
+// BoundedLabel is a type that represents a label and its associated
+// valid values
+type BoundedLabel struct {
+	LabelName   string
+	LabelValues []string
+}
+
+// getLabelNames extracts the all the `LabelName` fields from each BoundedLabel struct
+func getLabelNames(bl []BoundedLabel) []string {
+	ln := make([]string, 0)
+	for _, l := range bl {
+		ln = append(ln, l.LabelName)
+	}
+	return ln
+}
+
+func verifyBoundedLabels(bl []BoundedLabel) bool {
+	if len(bl) == 0 {
+		return false
+	}
+	for _, l := range bl {
+		if l.LabelValues == nil {
```

Comment:
More robust check here is `if len(l.labelValues) == 0`. Works for `nil` or if caller initializes an empty list.

Reply:
Sure, thanks a lot for the code review!

Reply:
Sure, thank you!